test/controllers/shared/basic_rest_response_test.rb

```diff
@@ -36,5 +36,108 @@ def basic_edit_test(object_found = nil)
+        end
+
+        test 'should render per page dropdown with correct values' do
+          assert_select "[id='per_page']" do
+            assert_select "option[selected='selected'][value='#{entries_per_page + 1}']"
+            assert_select "option[value='5']"
+            assert_select "option[value='10']"
+            assert_select "option[value='15']"
+            assert_select "option[value='25']"
+            assert_select "option[value='50']"
+          end
+        end
+
+        test 'should display pagination buttons' do
+          get :index, {per_page: entries_per_page + 1}, set_session_user
+          assert_response :success
+          assert_select "[id=pagination]" do
+            assert_includes @response.body.squish, "pagination pagination-pf-back"
+            assert_includes @response.body.squish, "pagination pagination-pf-forward"
+          end
+        end
+
+        test 'sort links should include per page param' do
+          sort_links = css_select('thead a')
+          sort_links.each do |link|
+            assert_includes link['href'], "per_page=#{entries_per_page + 1}"
+          end
+        end
+      end
+    end
+
+    def basic_pagination_rendered_test
+      context 'GET #index' do
+        old = Setting[:entries_per_page]
+        setup do
+          factory_name = get_factory_name(@controller.resource_class)
+          FactoryGirl.create(factory_name) if @controller.resource_class.count.zero?
+          Setting[:entries_per_page] = @controller.resource_class.count
+        end
+
+        test 'should not render pagination' do
+          get :index, {}, set_session_user
+          assert_response :success
+          refute_includes @response.body, "id=pagination"
+        end
+
+        test 'should render pagination' do
+          factory_name = get_factory_name(@controller.resource_class)
+          create_model_instance(factory_name)
+          get :index, {}, set_session_user
+          assert_response :success
+          assert_select "form[id='pagination']"
+        end
+
+        test 'should not render pagination when no search results' do
+          @request.env['HTTP_REFERER'] = root_url
+          get :index, {search: "name='A98$bcD#67Ef*g"}, set_session_user
+          assert (@response.body.include? "No entries found") ||  @response.body.match(/You are being.*redirected/)
+        end
+
+        teardown do
+          Setting[:entries_per_page] = old
+        end
+      end
+    end
+  end
+
+  def get_factory_name(model)
+    model = model.to_s
+    return :trend_os if model == "Trend"
+    if model.include? "::"
+      model = model.split("::")
+      if model.first != "Subnet"
+        model = model.last
+      else
+        model = model.first
+      end
+    end
+    model.underscore.to_sym
+  end
+
+  def create_model_instance(factory_name)
+    factory_name = :subnet_ipv4 if factory_name == :subnet
```

Comment:
Would it be easier to return `:subnet_ipv4` in the `get_factory_name` helper for any `Subnet` model, or just use the generic `:subnet` factory?